packages/launchpad/src/global/GlobalProjectCard.vue

```diff
@@ -42,7 +40,7 @@ const icons = {
 // but I'm struggling to get the types within the tests
 // When GQL exists, I'll be able to pull in the shared types.
 const props = defineProps<{
-  project: { name: string, lastRun: number, lastRunStatus: string }
+  project: { name: string, lastRun: number, lastRunStatus: string, id: string }
```

Comment:
We can use a `fragment` and grab the types from there (which is powered by `nxs.gen.ts`.

Here's how: https://github.com/cypress-io/cypress/pull/18270

Reply:
Yep, ideally if the component references stuff from GraphQL it will contain a fragment:

```ts
gql`
fragment GlobalProjectCard on Project {
  id
  name
  path
}
`

const props = defineProps<{
  gql: GlobalProjectCardFragment
}>()
```

And then whatever parent component nests `<GlobalProjectCard />` will just nest that fragment

```graphql
fragment GlobalPage on App {
  projects {
    id
    ...GlobalProjectCard
}
```